app/src/main/java/com/paranoidandroid/journey/recommendations/adapters/RecommendationsListAdapter.java

```diff
@@ -152,11 +142,14 @@ private void configureFoursquareVenueViewHolder(RecommendationViewHolder vh, int
     @Override
     public void onAddBookmarkClickedAt(int position) {
         if (this.listener != null) {
-            if (items.get(position).isBookmarked()) {
+            Recommendation rec = items.get(position);
+            if (rec.isBookmarked()) {
                 this.listener.onRemoveBookmarkClicked(items.get(position), position);
             } else {
                 this.listener.onAddBookmarkClicked(items.get(position), position);
             }
+            // Set immediately. Will be undone if save fails.
+            rec.setBookmarked(!rec.isBookmarked());
```

Comment:
We are breaking encapsulation by directly modifying the recommendation object inside the adapter.